cborg/src/Codec/CBOR/Pretty.hs

```diff
@@ -161,45 +163,63 @@ pprint = do
   hexRep term
   str " "
   case term of
-    TkInt      i  TkEnd -> ppTkInt i
-    TkInteger  i  TkEnd -> ppTkInteger i
-    TkWord     w  TkEnd -> ppTkWord w
-    TkBytes    bs TkEnd -> ppTkBytes bs
-    TkBytesBegin  TkEnd -> ppTkBytesBegin
-    TkString   t  TkEnd -> ppTkString t
-    TkStringBegin TkEnd -> ppTkStringBegin
-    TkListLen  w  TkEnd -> ppTkListLen w
-    TkListBegin   TkEnd -> ppTkListBegin
-    TkMapLen   w  TkEnd -> ppTkMapLen w
-    TkMapBegin    TkEnd -> ppTkMapBegin
-    TkBreak       TkEnd -> ppTkBreak
-    TkTag      w  TkEnd -> ppTkTag w
-    TkBool     b  TkEnd -> ppTkBool b
-    TkNull        TkEnd -> ppTkNull
-    TkSimple   w  TkEnd -> ppTkSimple w
-    TkFloat16  f  TkEnd -> ppTkFloat16 f
-    TkFloat32  f  TkEnd -> ppTkFloat32 f
-    TkFloat64  f  TkEnd -> ppTkFloat64 f
-    TkEncoded  _  TkEnd -> ppTkEncoded
-    TkEnd               -> str "# End of input"
+    TkInt      i   TkEnd     -> ppTkInt i
+    TkInt64    i   TkEnd     -> ppTkInt64 i
+    TkInteger  i   TkEnd     -> ppTkInteger i
+    TkWord64   w   TkEnd     -> ppTkWord64 w
+    TkWord     w   TkEnd     -> ppTkWord w
+    TkBytes    bs  TkEnd     -> ppTkBytes bs
+    TkBytesBegin   TkEnd     -> ppTkBytesBegin
+    TkByteArray ba TkEnd     -> ppTkByteArray ba
+    TkUtf8ByteArray ba TkEnd -> ppTkUtf8ByteArray ba
+    TkString   t   TkEnd     -> ppTkString t
+    TkStringBegin  TkEnd     -> ppTkStringBegin
+    TkListLen  w   TkEnd     -> ppTkListLen w
+    TkListBegin    TkEnd     -> ppTkListBegin
+    TkMapLen   w   TkEnd     -> ppTkMapLen w
+    TkMapBegin     TkEnd     -> ppTkMapBegin
+    TkBreak        TkEnd     -> ppTkBreak
+    TkTag      w   TkEnd     -> ppTkTag w
+    TkTag64    w   TkEnd     -> ppTkTag64 w
+    TkBool     b   TkEnd     -> ppTkBool b
+    TkNull         TkEnd     -> ppTkNull
+    TkUndef        TkEnd     -> ppTkUndef
+    TkSimple   w   TkEnd     -> ppTkSimple w
+    TkFloat16  f   TkEnd     -> ppTkFloat16 f
+    TkFloat32  f   TkEnd     -> ppTkFloat32 f
+    TkFloat64  f   TkEnd     -> ppTkFloat64 f
+    TkEncoded  _   TkEnd     -> ppTkEncoded
+    TkEnd                    -> str "# End of input"
     _ -> fail $ unwords ["pprint: Unexpected token:", show term]
```

Comment:
I thought we were going to remove the `_` catch-all case, since that's exactly what hid this problem in the first place. What cases is that catch-all covering? We can have specific failure cases, that's fine.